### ISSUE TITLE
Manual J HVAC sizing

### ISSUE THREAD

**Issue by shorowit** [NREL/OpenStudio-BEopt#8]
Implement Manual J sizing.

- [x] Add design conditions to weather.rb
- [x] Remove airflow rate inputs from HVAC measures
- [x] Calculation of heating/cooling/dehumidification loads
- [x] Calculation of duct loads
- [x] Equipment sizing for all types of HVAC equipment; update OS objects; unit tests
  - [x] Furnaces
  - [x] Central ACs
  - [x] ASHPs
  - [x] MSHPs
  - [x] Room ACs
  - [x] Electric Baseboard
  - [x] Boilers
  - [x] GSHPs
  - [x] Dehumidifiers
- [x] Remove DDY stuff from Location measure; change missing DDY to a warning instead of an error.
- [x] Remove design days from schedules
- [x] Remove airflow code from HVAC measures? (Probably. Eliminates duplicate code.)
- [x] Add runner.finalConditions
- [x] Unit tests for errors (e.g., missing construction, missing required unit feature, etc.)
- [x] FIXMEs 
- [x] AC/ASHP Capacity derate factors
- [x] Connect to UnitarySystemPerformance:Multispeed object

**Comment by ejhw** [comment]
FYI I confirmed that in BEopt we're using the EPW design data, and if it doesn't exist we calculate the 99%/1%/2% conditions:
```
    def _calc_design_info(self, hd, altitude):
        '''
        Calculate design day info: 
        - Heating 99% drybulb
        - Heating mean coincident windspeed 
        - Cooling 99% drybulb
        - Cooling mean coincident windspeed
        - Cooling mean coincident wetbulb
        - Cooling mean coincident humidity ratio
        '''
        
        std_press = Psychrometrics.Pstd_fZ(altitude)
        annual_hd_sorted_by_db = sorted(hd, key=lambda x: x['db'])
        annual_hd_sorted_by_dp = sorted(hd, key=lambda x: x['dp'])
        
        # 1%/99%/2% values
        heat99per_db = annual_hd_sorted_by_db[88]['db']
        cool01per_db = annual_hd_sorted_by_db[8673]['db']
        dehum02per_dp = annual_hd_sorted_by_dp[8584]['dp']
```
So Joe could switch to using 99% instead of 99.6%, etc. in OS.
cc @shorowit @joseph-robertson @jonwinkler

**Comment by jonwinkler** [comment]
That would be more in line with Manual J, which uses 99% for heating and 1% for cooling.

**Comment by shorowit** [comment]
If we implemented Manual J sizing, it would help with:
- DSE (distribution system efficiency)
- Normalized duct leakage
- Living vs. finished basement airflow rate
- No longer needing DDY files (which aren't even available for every EPW)
- Reducing effort needed for E+ autosizing (we can get rid of summer/winter design days, sizing objects, sizing run, etc.)

**Comment by shorowit** [comment]
Work has commenced on a [Manual J based measure](https://github.com/NREL/OpenStudio-BEopt/tree/master/measures/ResidentialHVACSizing).

**Comment by shorowit** [comment]
Turns out we need to set some airflow rates in the `UnitarySystemPerformance:Multispeed` object but it's not wrapped (though it's automatically forward-translated with autosized fields).

@nmerket @kbenne

**Comment by shorowit** [comment]
Connected to UnitarySystemPerformance:Multispeed object in https://github.com/NREL/OpenStudio-BEopt/commit/f7dee31437c55a4813e314d1ec92386ca9e2c6b2.

Fixed a GSHP bug in https://github.com/NREL/OpenStudio-BEopt/commit/f7bd152d928e35dcbd5de9976591c6326c0c5142 now that we could double-check the gFunctions.

This is finally done!

**Comment by nmerket** [comment]
Yay!


On Jun 23, 2017, at 3:15 PM, Scott Horowitz <notifications@github.com<mailto:notifications@github.com>> wrote:


Connected to UnitarySystemPerformance:Multispeed object in f7dee31<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FNREL%2FOpenStudio-BEopt%2Fcommit%2Ff7dee31437c55a4813e314d1ec92386ca9e2c6b2&data=02%7C01%7Cnoel.merket%40nrel.gov%7Ce2342aeb1eea406ed6ae08d4ba7d011e%7Ca0f29d7e28cd4f5484427885aee7c080%7C1%7C0%7C636338493431675622&sdata=jPu83i%2Fbd69iKRG6JJslTJJiCOPvdX0mDoKvbfN1vw8%3D&reserved=0>.

Fixed a GSHP bug in f7bd152<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FNREL%2FOpenStudio-BEopt%2Fcommit%2Ff7bd152d928e35dcbd5de9976591c6326c0c5142&data=02%7C01%7Cnoel.merket%40nrel.gov%7Ce2342aeb1eea406ed6ae08d4ba7d011e%7Ca0f29d7e28cd4f5484427885aee7c080%7C1%7C0%7C636338493431675622&sdata=X3zJlC5F4Bi79xOp43dgWO86LFeDyFkzqz%2FFFmniru4%3D&reserved=0> now that we could double-check the gFunctions.

This is finally done!

-
